### ISSUE TITLE
[5.0] Rewrite Action to run in nodejs environment / JavaScript rewrite

### ISSUE THREAD

**Issue by stefanzweifel** [stefanzweifel/git-auto-commit-action#44]
As filed in #18, this Action currently doesn't run in a nodejs environment and does not have access to `node`. Therefore tools like `husky` currently can't be used.

I've thought about this issue for quite a while now and can see that it would be beneficial for users, if they could use the Action in any environment. (In addition, it would make it easier to add a test suite to the project).

However, I'm also not the biggest fan of converting this  Action into a JavaScript project. The code is currently quite simple and does it's job well (It's basically just a shell script with ~50 lines of code). Converting it into JS or even TS requires to add a build process and updating JS dependencies if they become a security thread.

My bandwith is currently limited, as I work on other projects. And as I mentioned, the Action already works for 80% of usecases.

---

For me or for other users who would like to make this change happen here are some notes:

- Write Action preferrably in JavaScript and not in TypeScript, so more users can contribute
- Use one of the following libraries to interact with Git
  - https://www.nodegit.org/
  - https://github.com/steveukx/git-js
- Make the process of building the JS file and creating a new release as easy as possible (maybe even use a GitHub Action for that)

**Comment by spawnia** [comment]
I like that this action is focused and does one thing well. Personally i would advocate for keeping it that way 👍

Chaining actions allows to compose more complex use cases, e.g. running `prettier` and committing the result.

**Comment by GenieTim** [comment]
With the risk of asking a dumb question: why is the way via node/js needed? Why can't just the `entrypoint.sh` be the entrypoint run by the GitHub Action in whatever environment the other steps are run in? I see the point of having the Dockercontainer as that way this action can also be used e.g. in Windows environment, but since Docker is not used anymore currently, it is not that clear to me why a wrapper program is needed.

**Comment by stefanzweifel** [comment]
@GenieTim There are no dumb questions. It's actually a valid point.

Previously, users mentioned that they couldn't run tools like [husky](https://github.com/typicode/husky) with the Action, as the old Dockercontainer didn't contain `node`. In #45 we ditched the Dockercontainer and started relying on the `node12`-image of GitHub.

I've kept the issue open, as I once thought of rewriting the Action into a JS/TS Action. It's currently just a shell script and running automated tests with [bats](https://github.com/sstephenson/bats) hasn't been easy (Only now the idea came to me that I could use [bash_shell_mock](https://github.com/capitalone/bash_shell_mock) to mock git in the tests).

I thought rewriting the Action in JavaScript would make testing easier and would prevent me from releasing a broken Action.
However, over the last few days, I've come to the conclusion that this isn't the way to go (dependency nightmare 🙃 ).

I will stick to shell/bash. The Action should be as simple as possible and do it's one thing well.

(Closing the issue, as it really isn't relevant anymore)
